tls/s2n_resume.h

```diff
@@ -75,7 +75,7 @@ int s2n_encrypt_session_ticket(struct s2n_connection *conn, struct s2n_stuffer *
 int s2n_decrypt_session_ticket(struct s2n_connection *conn, struct s2n_stuffer *from);
 int s2n_encrypt_session_cache(struct s2n_connection *conn, struct s2n_stuffer *to);
 int s2n_decrypt_session_cache(struct s2n_connection *conn, struct s2n_stuffer *from);
-int s2n_config_is_encrypt_decrypt_key_available(struct s2n_config *config);
+S2N_RESULT s2n_config_is_encrypt_decrypt_key_available(struct s2n_config *config, bool *out);
```

Comment:
Does this need the `out` parameter? Do we want to distinguish between failure and false? We didn't make that distinction before. 

If we don't care about the distinction, you could make this a straight switch to S2N_RESULT, no output parameter, and call it like
```
if (s2n_result_is_ok(s2n_config_assert_encrypt_decrypt_key_available(config))) {

}
```

If we do care about the distinction, then you should clean up this naming. "is" implies a bool return value. "s2n_config_get_encrypt_decrypt_key_availability" or "s2n_config_check_encrypt_decrypt_key_available" would be better. Also, "out" is not a useful variable name.

Reply:
My understanding is we want the distinction between 1: we found the key, 2: we did not find the key, and 3: something caused an error during the process, to provide better diagnose tool. And yes I will change the naming.